phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java

```diff
@@ -230,14 +233,27 @@ public void testSkipDataTableAndPostIndexPartialRowUpdate() throws Exception {
         conn.commit();
         IndexRegionObserver.setSkipDataTableUpdatesForTesting(false);
         IndexRegionObserver.setSkipPostIndexUpdatesForTesting(false);
-        conn.createStatement().execute("upsert into " + dataTableName + " (id, val3) values ('a', 'abcdd')");
-        conn.commit();
         String selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
         // Verify that we will read from the first index table
         assertExplainPlan(conn, selectSql, dataTableName, indexName + "1");
         ResultSet rs = conn.createStatement().executeQuery(selectSql);
         assertTrue(rs.next());
         assertEquals("abc", rs.getString(1));
+        assertEquals("abcd", rs.getString(2));
+        assertFalse(rs.next());
+        selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val2  = 'abcc'";
+        // Verify that we will read from the second index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName + "2");
+        rs = conn.createStatement().executeQuery(selectSql);
+        assertFalse(rs.next());
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val3) values ('a', 'abcdd')");
+        conn.commit();
+        selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
+        // Verify that we will read from the first index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName + "1");
+        rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("abc", rs.getString(1));
         assertEquals("abcdd", rs.getString(2));
```

Comment:
Should not this value be abcd? Because the second and 3rd phase is failing?

Reply:
They fail only for one upsert and then the behavior is reverted back to normal at line 235 after
IndexRegionObserver.setSkipDataTableUpdatesForTesting(false);
IndexRegionObserver.setSkipPostIndexUpdatesForTesting(false);
You need to look at the file not just the diff to see it all

Reply:
@gjacoby126, I added more testing to cover the case where back to back data table row failures happen. Other failures cases were covered in GlobalIndexCheckerIT before. PHOENIX-5055 is to handle data write failures during overwrites.